frontend/src/main/scala/bloop/engine/Interpreter.scala

```diff
@@ -110,20 +110,27 @@ object Interpreter {
       f: State => Task[State]
   ): Task[State] = {
     val reachable = Dag.dfs(getProjectsDag(projects, state))
-    val allSources = reachable.iterator.flatMap(_.sources.toList).map(_.underlying)
-    val watcher = SourceWatcher(projects.map(_.name), allSources.toList, state.logger)
-    val fg = (state: State) => {
-      val newState = State.stateCache.getUpdatedStateFrom(state).getOrElse(state)
-      f(newState).map { state =>
-        watcher.notifyWatch()
-        State.stateCache.updateBuild(state)
-      }
-    }
+    val projectsSourcesAndDirs = reachable.map(_.allSourceFilesAndDirectories)
+    Task
+      .gatherUnordered(projectsSourcesAndDirs)
```

Comment:
Can we use `Task.gatherN` (assuming it's availble in our Monix version)? I'd like to avoid submitting to the IO pool a lot of tasks that cause the pool to allocate a lot of threads at once

Reply:
We use monix 2.3.3, where it's not available.

Reply:
Can we use something like this?

```
      val groupTasks = projectSourcesAndDirs.grouped(8).map(group => Task.gatherUnordered(group)).toList
      Task.sequence(groupTasks).map(fp => fp.flatten)
              ...
```

It's the normal pattern we use in the codebase today, in absences of `Task.gatherN`

The 8 number is totally arbitrary but given that this is IO it's fine?